### ISSUE TITLE
Run type-generation at runtime

### ISSUE THREAD

**Issue by Weakky** [graphql-nexus/nexus-plugin-prisma#309]
## Description

Currently, both the _library_ generation and the _types_ generation are done at the same time (when running `prisma2 generate`).

To enable further customization of the generated schema and the typings, we're thinking about moving the type-generation at runtime (just like) nexus, to prevent from having all the config inputted in the generator config.

The nexus-prisma generator should generate types only if none exists yet. Otherwise, it should be handled by the `nexusPrismaPlugin` itself.

## Use-cases

One use-case could be, for instance, the customization of CRUD field names:

> Note: The API design below is purely made up.

```ts
import { nexusPrismaPlugin } from '@generated/nexus-prisma'

const nexusPrisma = nexusPrismaPlugin({
  photon: ctx => ctx.photon,
  fieldNames: {
    Query: {
      findMany: (typeName) => `${typeName.toLowerCase()}s`,
      findOne:  (typeName) => `${typeName.toLowerCase()}`
    }
  },
})
```

The `fieldNames` property could then be both used to modify the runtime behavior but also the type-generation so that the typings reflect the configuration:

(Given a `User` prisma model)

```ts
objectType({
  name: 'Query',
  definition(t) {
    t.crud.users() // findMany
	t.crud.user()  // findOne
  }
})
```

**Comment by jasonkuhrt** [comment]
@Weakky I think we can consider this shipped thanks to #405.

**Comment by Weakky** [comment]
Agreed, let's close it!
